hazelcast/src/main/java/com/hazelcast/config/SerializationConfig.java

```diff
@@ -454,6 +455,20 @@ public SerializationConfig setAllowUnsafe(boolean allowUnsafe) {
         return this;
     }
 
+    /**
+     * @return true if embedded serializers may be overridden by custom serializers
+     *
+     * @since 4.2
+     */
+    public boolean isAllowOverrideDefaultSerializers() {
+        return allowOverrideDefaultSerializers;
+    }
+
+    public SerializationConfig setAllowOverrideDefaultSerializers(final boolean allowOverrideDefaultSerializers) {
```

Comment:
Can we add something like following into Javadoc of the set method?
```
This configuration should be used cautiously. Default serializers are used heavily by the hazelcast internally. 
If any of the instance in the same cluster overrides a default serializer, all the members and clients in the same cluster must override it with the same serializer. 
This configuration is introduced specifically to support the following case:
1. There was a custom configuration by the user for a type 
2. Hazelcast decided to add a default serializer for the same type in a future release.
To be able to support Rolling Upgrade from an old version to the new version, one needs to make sure of the backward compatibility of the serialization. As a solution, the user will set this property on the new version so that new instances will be able to override new default serializers with their existing custom serializers. 
```

Reply:
Javadoc added